securedrop/source_app/__init__.py

```diff
@@ -106,6 +107,15 @@ def check_tor2web() -> Optional[werkzeug.Response]:
             return redirect(url_for("info.tor2web_warning"))
         return None
 
+    @app.before_request
```

Comment:
I originally added this as a single check at startup and added a config value to refer to instead, just to avoid the overhead here. We could probably do it on the index page only (since you can't get far in the app without signing up or logging in from there) but per-request seems more bulletproof so long as it's simple. I wouldn't do it per-request if we're checking the key every time.

Reply:
Oops, I thought I was simplifying. OK, I've re-added that back in, but I'm going to try to wait for the staging test to complete before pushing again.